clang/lib/AST/DeclCXX.cpp

```diff
@@ -826,6 +842,14 @@ void CXXRecordDecl::addedMember(Decl *D) {
               ? !Constructor->isImplicit()
               : (Constructor->isUserProvided() || Constructor->isExplicit()))
         data().Aggregate = false;
+
+      // A trivially relocatable class is a class:
+      // -- where no eligible copy constructor, move constructor, copy
+      // assignment operator, move assignment operator, or destructor is
+      // user-provided,
+      if (Constructor->isUserProvided() && (Constructor->isCopyConstructor() ||
+                                            Constructor->isMoveConstructor()))
+        data().IsNaturallyTriviallyRelocatable = false;
```

Comment:
Where did you get that definition from?
In P2786R4, `C` is trivial if `C(someC())`  does not invoke a user-defined constructor, which is more corect vis a vis constraints. (So that would have to be done from outside the class, Maybe in Sema)

Reply:
If there is a user-defined copy constructor, then it's not trivially copyable anymore. Same goes for moves. So P1144 is more in line with what the standard has chosen in the past. IMO it's better to keep the same principle and not hide things to the developer.

Reply:
Consider, for example
```
template <typename T>
struct S {
    S(const S&) = default;
    S(const S&) requires false = delete;
};
```

https://compiler-explorer.com/z/5s5rja6d3